source-code/app/src/main/java/org/buildmlearn/toolkit/ToolkitApplication.java

```diff
@@ -20,7 +20,7 @@
     private String toolkitDir;
     private String apkDir;
 
-    private String dir;
+    private static String dir;
```

Comment:
Why static? Memory leak will occur .... can you try implementing with `non static`?